eskip/parser.y

```diff
@@ -98,13 +98,14 @@ routes:
 	}
 
 routedef:
-	routeid colon route {
-		$$.route = $3.route
+	routeid route {
+		$$.route = $2.route
 		$$.route.id = $1.token
 	}
 
 routeid:
-	symbol {
+	symbol colon {
+		// match symbol and colon to get route id early even if route parsing fails later
```

Comment:
Is there any performance improvement here?

Reply:
No, performance change is within measurement deviation
```
goos: linux
goarch: amd64
pkg: github.com/zalando/skipper/eskip
cpu: Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
                  │   HEAD~1    │                HEAD                │
                  │   sec/op    │   sec/op     vs base               │
ParsePredicates-8   8.189µ ± 4%   8.515µ ± 4%  +3.98% (p=0.007 n=10)
Parse-8             253.3m ± 2%   256.8m ± 3%  +1.38% (p=0.011 n=10)
geomean             1.440m        1.479m       +2.67%

                  │    HEAD~1    │                HEAD                 │
                  │     B/op     │     B/op      vs base               │
ParsePredicates-8   1.961Ki ± 0%   1.961Ki ± 0%       ~ (p=1.000 n=10)
Parse-8             49.02Mi ± 0%   49.02Mi ± 0%       ~ (p=0.566 n=10)
geomean             313.7Ki        313.7Ki       -0.00%

                  │   HEAD~1    │                 HEAD                 │
                  │  allocs/op  │  allocs/op   vs base                 │
ParsePredicates-8    32.00 ± 0%    32.00 ± 0%       ~ (p=1.000 n=10) ¹
Parse-8             1.080M ± 0%   1.080M ± 0%       ~ (p=0.700 n=10)
geomean             5.879k        5.879k       +0.00%
¹ all samples are equal
```